lib/cpp/src/thrift/server/TThreadedServer.h

```diff
@@ -20,120 +20,86 @@
-                  THRIFT_OVERLOAD_IF(ProcessorFactory, TProcessorFactory));
+                  const boost::shared_ptr<transport::TServerTransport>& serverTransport,
+                  const boost::shared_ptr<transport::TTransportFactory>& transportFactory,
+                  const boost::shared_ptr<protocol::TProtocolFactory>& protocolFactory,
+                  const boost::shared_ptr<concurrency::ThreadFactory>& threadFactory,
+                  THRIFT_OVERLOAD_IF(ProcessorFactory, TProcessorFactory))
+    : TServer(processorFactory, serverTransport, transportFactory, protocolFactory),
+      threadFactory_(threadFactory),
+      stop_(false) {}
 
   template <typename Processor>
   TThreadedServer(const boost::shared_ptr<Processor>& processor,
-                  const boost::shared_ptr<TServerTransport>& serverTransport,
-                  const boost::shared_ptr<TTransportFactory>& transportFactory,
-                  const boost::shared_ptr<TProtocolFactory>& protocolFactory,
-                  THRIFT_OVERLOAD_IF(Processor, TProcessor));
+                  const boost::shared_ptr<transport::TServerTransport>& serverTransport,
+                  const boost::shared_ptr<transport::TTransportFactory>& transportFactory,
+                  const boost::shared_ptr<protocol::TProtocolFactory>& protocolFactory,
+                  THRIFT_OVERLOAD_IF(Processor, TProcessor))
+    : TServer(processor, serverTransport, transportFactory, protocolFactory),
+      threadFactory_(new concurrency::PlatformThreadFactory),
+      stop_(false) {}
 
   template <typename Processor>
   TThreadedServer(const boost::shared_ptr<Processor>& processor,
-                  const boost::shared_ptr<TServerTransport>& serverTransport,
-                  const boost::shared_ptr<TTransportFactory>& transportFactory,
-                  const boost::shared_ptr<TProtocolFactory>& protocolFactory,
-                  const boost::shared_ptr<ThreadFactory>& threadFactory,
-                  THRIFT_OVERLOAD_IF(Processor, TProcessor));
+                  const boost::shared_ptr<transport::TServerTransport>& serverTransport,
+                  const boost::shared_ptr<transport::TTransportFactory>& transportFactory,
+                  const boost::shared_ptr<protocol::TProtocolFactory>& protocolFactory,
+                  const boost::shared_ptr<concurrency::ThreadFactory>& threadFactory,
+                  THRIFT_OVERLOAD_IF(Processor, TProcessor))
+    : TServer(processor, serverTransport, transportFactory, protocolFactory),
+      threadFactory_(threadFactory),
+      stop_(false) {}
 
-  virtual ~TThreadedServer();
+  virtual ~TThreadedServer() {};
 
   virtual void serve();
 
-  void stop() {
+  virtual void stop() {
     stop_ = true;
     serverTransport_->interrupt();
   }
 
 protected:
-  void init();
+  virtual void disposeClient(TConnectedClient *pClient);
 
-  boost::shared_ptr<ThreadFactory> threadFactory_;
+  boost::shared_ptr<concurrency::ThreadFactory> threadFactory_;
```

Comment:
I also think it would make sense to follow best practices on the using clauses and remove all of them from all headers so that namespaces are not contaminated outside the project.  This makes for longer declarations in headers but it is far more reliable especially when switching language revisions or running into things that you mentioned with Microsoft ::concurrency.